### PR TITLE
Search adding

### PR DESCRIPTION
**Pull request by esidarapp**

### CODE REVIEW

File: src/main/java/com/example/spring/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false,unique = true)
+    private String isbn;
+    @Column(nullable = false)
+    private BigDecimal price;
+    private String description;
+    @Column(name = "cover_image")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/model/Book.java]:
```suggestion

```
Hibernate automatically create name with snake_case

File: src/main/java/com/example/spring/repository/book/AuthorSpecificationProvider.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import com.example.spring.model.Book;
+import com.example.spring.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "author";
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/repository/book/AuthorSpecificationProvider.java]: Create constant for "author"

File: src/main/java/com/example/spring/repository/book/IsbnSpecificationProvider.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import com.example.spring.model.Book;
+import com.example.spring.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class IsbnSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "isbn";
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/repository/book/IsbnSpecificationProvider.java]: the same

File: src/main/java/com/example/spring/repository/book/TitleSpecificationProvider.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import com.example.spring.model.Book;
+import com.example.spring.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "title";
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/repository/book/TitleSpecificationProvider.java]: the same

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1 +1,16 @@
...
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+
+spring.jpa.show-sql=true
+spring.jpa.hibernate.ddl-auto=validate
+spring.jpa.open-in-view=false
+
+spring.jackson.deserialization.fail-on-unknown-properties=true
+
+spring.liquibase.changelog=classpath:db/changelog/db.changelog-master.yaml
+
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties]: A lot of blank lines at the end, leave just one

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: tinyint(1)
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml:49]:
```suggestion
                    nullable: false

```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/db.changelog-master.yaml:3]:
```suggestion
      file: db/changelog/changes/01-create-books-table.yaml

```

File: src/main/java/com/example/spring/repository/BookSpecificationBuilder.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package com.example.spring.repository;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/repository/BookSpecificationBuilder.java]: Let's create package filter for example, and replcae all searching logic there

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it!

File: src/main/java/com/example/spring/service/BookServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        );
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book "
+                        + "by id " + id)
+        );
```
Comment (okuzan) [inline src/main/java/com/example/spring/service/BookServiceImpl.java]: repeated logic, extract to a private method

File: src/main/java/com/example/spring/service/BookServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        bookRepository.save(book);
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
```
Comment (okuzan) [inline src/main/java/com/example/spring/service/BookServiceImpl.java:35]: btw you can do it in Mapstruct mapper to - just declare list to list mapping

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**